spec/cddl/many.cddl

```diff
@@ -39,6 +39,11 @@ request = COSE_Sign<{
     ; Attributes. An optional list of request attributes defined in the spec. See
     ; the list of all attributes in this repo.
     ? 8 => [ * attribute ],
+
+    ; Flag.  When set, a proof of what was requested will be given, if possible.
```

Comment:
I think an attribute here would be better, as:

- Attributes are meant to be optionally implemented, with clear rules when implementation is missing (request handlers reject them, response handlers ignore them) which actually makes sense here,
- Adding fields to messages is a bit like releasing a new version of HTTP. Attributes are the mechanism to enhance them (similar to HTTP headers[1]),
- Servers can advertise which attributes they support, but with this ALL servers should implement proofs, which is unreasonable.

So I'd rather have a system similar to `async` support; a request attribute, a response attribute and a server attribute that indicates support for those. This way servers can advertise they support proofs, clients can ignore proofs, and if proof is requested the server should bail out as it cannot provide it, and client can then re-request without the attribute if it chooses so.

[1] there is an argument that HTTP headers are different as they are optional while attributes require handling or fail, but in this specific case it is actually beneficial.

Reply:
Acknowledged . . . will refactor this into a proof attribute